src/Css/Style.php

```diff
@@ -1361,6 +1394,12 @@ protected function clear_cache(string $prop): void
      */
     public function set_prop(string $prop, $val, bool $important = false, bool $clear_dependencies = true): void
     {
+        // Store CSS custom property as-is and leave.
+        if (substr($prop, 0, 2) === "--") {
+            $this->_props[$prop] = trim($val);
+            return;
+        }
```

Comment:
In the lines below this block, all dashes are converted to underscores. This would need to get an exception too because `--custom-var` and `--custom_var` are both valid, but are two different properties. I am not sure if the getters accept dashes.

Reply:
The purpose of swapping out dash for underscore is so that the getter/setter can be used at all, since dashes aren't valid identifiers. In this way we can get the `font-face` value by accessing the font_face property (`$style->font_face`).

But you're right that both dash and underscore are permitted since the custom property name is defined as a [dashed identity](https://drafts.csswg.org/css-values-4/#typedef-dashed-ident). So, skipping replacement makes sense. And any "get" would just have to use the less convenient `$style->__get("--custom-prop");`.

I do still think we should continue processing rather than set and return, with some additional handling where needed.

Reply:
I think this is resolved. 

        // Skip some checks for CSS custom properties.
        if (substr($prop, 0, 2) !== "--") {

            $prop = str_replace("-", "_", $prop);

            // Legacy property aliases
            if (isset(self::$_props_alias[$prop])) {
                $prop = self::$_props_alias[$prop];
            }

            if (!isset(self::$_defaults[$prop])) {
                global $_dompdf_warnings;
                $_dompdf_warnings[] = "'$prop' is not a recognized CSS property.";
                return;
            }
        }